### ISSUE TITLE
More comment related issues

### ISSUE THREAD

**Issue by zpao** [babel/babel-eslint#153]
In a similar vein as fd384851e0c93f3e9ee6547361fd9b8649f886a9, with latest babel-core and babel-eslint, the following code is having issues (actual reduced test case this time!):

```js
var a = [
  1,
  2, // a trailing comment makes this line fail comma-dangle (always-multiline)
];

switch (a) {
  // A comment here makes the above line fail brace-style
  case 1:
    console.log(a);
}
```

(here's our eslintrc which I'm running this against: https://github.com/facebook/react/blob/master/.eslintrc)

**Comment by sebmck** [comment]
I can't seem to replicate this:

```
$ cat .eslintrc
---
parser: babel-eslint

plugins:
  - react

env:
  browser: true
  node: true

globals:
  __DEV__: true
  # Jest / Jasmine
  describe: false
  xdescribe: false
  beforeEach: false
  afterEach: false
  it: false
  xit: false
  jest: false
  pit: false
  expect: false
  spyOn: false
  jasmine: false

rules:
  # ERRORS
  space-before-blocks: 2
  indent: [2, 2, indentSwitchCase: true]
  brace-style: 2
  space-after-keywords: 2
  strict: 2
  comma-dangle: [2, always-multiline]
  # Make this a warning for now. We do this in a few places so we might need to
  # disable
  no-unused-expressions: 2
  eol-last: 2
  dot-notation: 2
  dot-location: [2, property]
  consistent-return: 2
  no-unused-vars: [2, args: none]
  quotes: [2, single, avoid-escape]
  no-shadow: 2
  no-multi-spaces: 2

  # WISHLIST. One day...
  # We'll need a custom version of this that does a subset of the whole rule.
  # Otherwise this is just too noisy.
  # valid-jsdoc: 1
  # Ideally, we could just warn when *new* lines are added that exceed 80 chars
  # while not warning about existing ones (often URLs, etc. which are
  # necessarily long), but we don't have a good way to do so.
  # max-len: [0, 80]

  # DISABLED. These aren't compatible with our style
  # We use this for private/internal variables
  no-underscore-dangle: 0
  # We pass constructors around / access them from members
  new-cap: 0
  # We do this a lot.
  no-use-before-define: 0
  # We do this in a few places to align values
  key-spacing: 0
  # It's nice to be able to leave catch blocks empty
  no-empty: 0
  # It makes code more readable to make this explicit sometimes
  no-undef-init: 0

  # BROKEN. We'd like to turn these back on.
  # causes a ton of noise, eslint is too picky?
  block-scoped-var: 0

  # JSX
  # Our transforms set this automatically
  react/display-name: 0
  react/jsx-boolean-value: [2, always]
  react/jsx-no-undef: 2
  react/jsx-quotes: [2, double]
  # We don't care to do this
  react/jsx-sort-prop-types: 0
  react/jsx-sort-props: 0
  react/jsx-uses-react: 2
  react/jsx-uses-vars: 2
  # It's easier to test some things this way
  react/no-did-mount-set-state: 0
  react/no-did-update-set-state: 0
  # We define multiple components in test files
  react/no-multi-comp: 0
  react/no-unknown-property: 2
  # This isn't useful in our test code
  react/prop-types: 0
  react/react-in-jsx-scope: 2
  react/self-closing-comp: 2
  # We don't care to do this
  react/sort-comp: 0
  react/wrap-multilines: [2, {declaration: false, assignment: false}]
```

```
$ cat index.js
var a = [
  1,
  2, // a trailing comment makes this line fail comma-dangle (always-multiline)
];

switch (a) {
  // A comment here makes the above line fail brace-style
  case 1:
    console.log(a);
}
```

```
$ eslint index.js
```

**Comment by sebmck** [comment]
Running `eslint .` on React master also doesn't report any other errors.

**Comment by zpao** [comment]
React master wouldn't since it's still on babel-core@5.6.x. I just made https://github.com/facebook/react/pull/4456 which hopefully acts on travis like it does locally (2 classes of failures in the test case)

**Comment by sebmck** [comment]
Ah, I had `babel-eslint` locally linked but didn't have `babel-core`. Should be a fairly easy fix, only a couple of things this could be.

**Comment by sebmck** [comment]
Fixed as of 3.1.25, thanks! Comment tokens were being included in the token list which ESLint didn't like.

**Comment by zpao** [comment]
Thanks! You're the best :)

**Comment by hzoo** [comment]
@sebmck To fix the tests I also used the filtered tokens list
`exports.attachComments(ast, comments, ast.tokens);` instead of
`exports.attachComments(ast, comments, tokens);`
